crescent_bot/plugins/info.py

```diff
@@ -19,33 +19,36 @@ class UserInfo:
     async def callback(self, ctx: crescent.Context):
         # If a user uses a user id, they can select a user that isn't
         # in the guild.
-        if not isinstance(self.target, hikari.Member):
-            await ctx.respond("That user is not in the server.")
+        if not isinstance(self.target, hikari.InteractionMember):
+            await ctx.respond("That user is not in this server.")
             return
 
         created_at = int(self.target.created_at.timestamp())
         joined_at = int(self.target.joined_at.timestamp())
-        roles = (await self.target.fetch_roles())[1:]  # All but @everyone.
+
+        # Create a list of roles, formatting them into role mentions.
+    # The @everyone role's ID is the guild's ID, which we filter out.
+        roles = [f"<@&{role}>" for role in self.target.role_ids if role != ctx.guild_id]
 
         # Function calls can be chained when creating embeds.
         embed = (
             hikari.Embed(
                 title="User information",
-                description=f"ID: {self.target.id}",
+                description=f"ID: `{self.target.id}`",
                 colour=hikari.Colour(0x563275),
                 # Doing it like this is important.
                 timestamp=dt.datetime.now().astimezone(),
             )
-            .set_author(name="Information")
+            .set_author(name=str(self.target))
```

Comment:
Should this not use the `display_name` attribute?

Reply:
It'll show the user's username and discriminator, but I can use the display_name property if you'd like

Reply:
Ah okay. In that case it's fine as is.